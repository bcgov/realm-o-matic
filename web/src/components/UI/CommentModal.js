import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { Modal, Button, TextArea } from 'semantic-ui-react';
import { TEST_IDS } from '../../constants/ui';

const StyledModal = styled(Modal)`
  &&& {
    width: 400px;
    text-align: center;
    padding: 10px 0;
  }
  .content {
    margin: 30px 10px;
  }
  .rejectionMessage {
    margin: 10px;
    width: 300px;
    height: 200px;
    border: 1px solid #036;
  }
  .ui.button {
    color: white;
    background-color: #036;
  }
  .hidden {
    display: none !important;
  }
`;

/**
 * Pop up modal for comment
 * @param {Boolean} openRejectionModal Open modal when reject
 * @param {Function} onModalClose Action when closing the modal
 * @param {Boolean} disabled Text field editable or not
 * @param {Function} onMessageChange Action when updating message
 * @param {String} message The content to display when not editing
 */
export const CommentModal = ({
  openRejectionModal,
  onModalClose,
  disabled,
  onMessageChange,
  message = '',
}) => {
  // Submit button to submit a rejection message. If only viewing, hide the button:
  const buttons = (
    <div>
      <Button onClick={() => onModalClose(false)}>Close</Button>
      <Button onClick={() => onModalClose(true)} className={disabled ? 'hidden' : null}>
        Submit
      </Button>
    </div>
  );

  const onCommentChange = (e, { value }) => onMessageChange(value);

  // The comment text area, either editable or display mode:
  const comment = (
    <TextArea
      placeholder="Reason for Rejection"
      className="rejectionMessage"
      value={message}
      onChange={onCommentChange}
      disabled={disabled}
    />
  );

  const options = {
    open: openRejectionModal,
    closeOnEscape: true,
    closeOnDimmerClick: false,
    size: 'tiny',
    dimmer: 'blurring',
    header: 'BCeID Rejection Message',
    content: comment,
    actions: buttons,
  };
  return <StyledModal data-testid={TEST_IDS.REQUEST.REJECT_MODAL} {...options} />;
};

CommentModal.propTypes = {
  openRejectionModal: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired,
  message: PropTypes.string,
};

export default CommentModal;
