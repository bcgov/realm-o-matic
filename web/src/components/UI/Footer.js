import React from 'react';
import { Grid, Container } from 'semantic-ui-react';
import styled from '@emotion/styled';
import { FOOTER_LINKS } from '../../constants';

const StyledFooter = styled.footer`
  position: fixed;
  bottom: 0;
  width: 100%;
  background-color: #036;
  border-top: 2px solid #fcba19;
  color: #fff;
  font-size: 15px;
  padding: 10px 0;
  @media (max-width: 768px) {
    position: relative;
  }
`;

const columns = FOOTER_LINKS.map((link, index) => (
  <Grid.Column key={index} mobile={6} computer={2}>
    <a href={link.link}>{link.name}</a>
  </Grid.Column>
));

export const Footer = () => {
  return (
    <StyledFooter>
      <Container>
        <Grid>{columns}</Grid>
      </Container>
    </StyledFooter>
  );
};

export default Footer;
