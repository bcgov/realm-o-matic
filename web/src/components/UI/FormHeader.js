import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Button, Grid } from 'semantic-ui-react';
import styled from '@emotion/styled';
import { TEST_IDS } from '../../constants/ui';

const StyledFormHeader = styled.div`
  .hidden {
    display: none !important;
  }

  color: #036;

  margin-bottom: 10px;

  .column {
    margin-top: 15px !important;
  }

  .ui.positive.button {
    color: white;
    background-color: #036;
  }

  .ui.button {
    color: #036;
    background: none;
    border-radius: 4px;
    border: 1px solid #003366 !important;
  }
`;

/**
 * Generate header for form with actions
 * @param {String} title the title of form
 * @param {Boolean} hideAction if the action buttons should be hidden
 * @param {Boolean} hideRejectionViewButton if the rejection message buttons should be hidden
 * @param {Function} onApprove action on approval
 * @param {Function} onReject action on rejection
 * @param {Function} onViewRejection action on viewing rejection message
 */
export const FormHeader = ({
  title = 'Form',
  hideAction = true,
  onApprove,
  onReject,
  hideRejectionViewButton = true,
  onViewRejection,
}) => {
  return (
    <StyledFormHeader>
      <Grid>
        <Grid.Row>
          <Grid.Column mobile={16} tablet={8} computer={3}>
            <Link to="/home" data-testid={TEST_IDS.REQUEST.CANCEL}>
              <Button>Back</Button>
            </Link>
          </Grid.Column>
          <Grid.Column mobile={16} tablet={8} computer={9}>
            <h2>{title}</h2>
          </Grid.Column>
          <Grid.Column mobile={16} tablet={8} computer={4}>
            <Button.Group className={hideAction ? 'hidden' : null}>
              <Button onClick={onReject} data-testid={TEST_IDS.REQUEST.REJECT}>
                Reject
              </Button>
              <Button.Or />
              <Button positive onClick={onApprove} data-testid={TEST_IDS.REQUEST.APPROVAL}>
                Approve
              </Button>
            </Button.Group>
            <Button
              className={hideRejectionViewButton ? 'hidden' : null}
              onClick={onViewRejection}
              data-testid={TEST_IDS.REQUEST.VIEW_REJECT}
            >
              Rejection Message
            </Button>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </StyledFormHeader>
  );
};

FormHeader.propTypes = {
  title: PropTypes.object,
  hideAction: PropTypes.bool,
  onApprove: PropTypes.func,
  onReject: PropTypes.func,
  onViewRejection: PropTypes.func,
};

export default FormHeader;
