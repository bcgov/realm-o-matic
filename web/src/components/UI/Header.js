import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Menu, Container, Dropdown, Icon } from 'semantic-ui-react';
import styled from '@emotion/styled';
/** @jsx jsx */
import { css, jsx } from '@emotion/core';
import { APP_INFO } from '../../constants/ui';
import logo from '../../assets/bcgovlogo.svg';
import { TEST_IDS } from '../../constants/ui';

const StyledHeader = styled(Menu)`
  &&& {
    background: #036;
    border-bottom: 2px solid #fcba19;
    font-size: 18px;
  }
`;

const StyledItem = styled(Menu.Item)`
  &&& {
    font-size: 1rem;
    padding: 0;
    margin: 0;
  }
`;

const StyledImage = styled.img`
  padding: 0;
  margin: 5px 0 !important;
  width: 80px !important;
`;

const DropdownCss = css`
  padding: 35px 0 !important;
  color: white !important;
`;

const TextCss = css`
  font-size: 22px;
  margin-left: 15px !important;
  color: white !important;
`;

// TODO: fix mobile view
export const Header = ({ authentication }) => {
  const { isAuthenticated, userInfo } = authentication;

  const trigger = (
    <span>
      <Icon name="user" /> Account
    </span>
  );

  const authenticationState = {
    key: isAuthenticated ? 'signout' : 'signin',
    text: isAuthenticated ? 'Sign Out' : 'Sign In',
    as: Link,
    to: isAuthenticated ? '/logout' : '/login',
  };

  const options = [
    {
      key: 'user',
      text: (
        <span>
          Signed in as <strong>{userInfo ? userInfo.username : 'NONAME'}</strong>
        </span>
      ),
      disabled: true,
    },
    { key: 'requests', text: 'Requests', as: Link, to: '/home' },
    authenticationState,
  ];

  const AccountDropdown = (
    <Dropdown
      trigger={trigger}
      options={options}
      css={DropdownCss}
      data-testid={TEST_IDS.APP.ACCOUNT}
    />
  );

  return (
    <StyledHeader fixed="top">
      <Container>
        <StyledItem>
          <StyledImage src={logo} alt="logo" data-testid={TEST_IDS.APP.LOGO} />
        </StyledItem>
        <StyledItem>
          <Link to={{ pathname: '/' }} css={TextCss} data-testid={TEST_IDS.APP.NAME}>
            {APP_INFO.NAME}
          </Link>
        </StyledItem>
        <StyledItem position="right">{AccountDropdown}</StyledItem>
      </Container>
    </StyledHeader>
  );
};

Header.propTypes = {
  authentication: PropTypes.object.isRequired,
};

export default Header;
