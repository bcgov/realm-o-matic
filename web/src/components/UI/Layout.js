import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter } from 'react-router-dom';
import { Container } from 'semantic-ui-react';
import styled from '@emotion/styled';
import { Header, Footer } from './';

const StyledApp = styled.div`
  text-align: center;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`;

export const Layout = ({ children }) => {
  return (
    <BrowserRouter>
      <StyledApp>
        <Header />
        <Container>{children}</Container>
        <Footer />
      </StyledApp>
    </BrowserRouter>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};
