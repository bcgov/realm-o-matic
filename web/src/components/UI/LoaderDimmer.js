import React from 'react';
import PropTypes from 'prop-types';
import { Dimmer, Segment } from 'semantic-ui-react';
import { SpinLoader } from './SpinLoader';

/**
 * Page dimmer with a spin loader
 * @param {String} text the text to pass to loader
 * @param {Boolean} idDim if true then show dimmer
 */
export const LoaderDimmer = ({ text, idDim, children }) => {
  return (
    <Segment>
      <Dimmer active={idDim} page blurring="true">
        <SpinLoader text={text} />
      </Dimmer>
      {children}
    </Segment>
  );
};

LoaderDimmer.propTypes = {
  text: PropTypes.string,
  idDim: PropTypes.bool,
  children: PropTypes.node,
};

export default LoaderDimmer;
