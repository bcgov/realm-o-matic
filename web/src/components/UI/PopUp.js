import React from 'react';
import PropTypes from 'prop-types';
import { Popup, Icon } from 'semantic-ui-react';
import { TEST_IDS } from '../../constants/ui';

/**
 * Popup with color and text
 * @param {Object} status the status of the form
 */
export const PopUp = ({ status }) => {
  return (
    <Popup
      content={`Status: ${status.text}`}
      trigger={
        <Icon name="circle" color={status.color} size="tiny" data-testid={TEST_IDS.APP.POPUP} />
      }
    />
  );
};

PopUp.propTypes = {
  status: PropTypes.object,
};

export default PopUp;
