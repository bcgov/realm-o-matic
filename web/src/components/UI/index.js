export { default as Header } from './Header';
export { Footer } from './Footer';
export { Layout } from './Layout';
