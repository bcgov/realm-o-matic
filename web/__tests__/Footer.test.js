import React from 'react';
import { render } from 'react-testing-library';
import { Footer } from '../src/components/UI/Footer';

describe('Footer Component', () => {
  it('renders without crashing', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
