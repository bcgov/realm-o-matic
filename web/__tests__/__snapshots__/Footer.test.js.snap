// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders without crashing 1`] = `
<div>
  <footer
    class="css-1r34j8h-StyledFooter ewud6l10"
  >
    <div
      class="ui container"
    >
      <div
        class="ui grid"
      >
        <div
          class="two wide computer six wide mobile column"
        >
          <a
            href="."
          >
            Home
          </a>
        </div>
        <div
          class="two wide computer six wide mobile column"
        >
          <a
            href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
          >
            Disclaimer
          </a>
        </div>
        <div
          class="two wide computer six wide mobile column"
        >
          <a
            href="https://www2.gov.bc.ca/gov/content/home/privacy"
          >
            Privacy
          </a>
        </div>
        <div
          class="two wide computer six wide mobile column"
        >
          <a
            href="https://www2.gov.bc.ca/gov/content/home/accessibility"
          >
            Accessibility
          </a>
        </div>
        <div
          class="two wide computer six wide mobile column"
        >
          <a
            href="https://www2.gov.bc.ca/gov/content/home/copyright"
          >
            Copyright
          </a>
        </div>
        <div
          class="two wide computer six wide mobile column"
        >
          <a
            href="https://github.com/bcgov/realm-o-matic"
          >
            Contact Us
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
